packages/dd-trace/src/format.js

```diff
@@ -124,43 +129,48 @@ function extractTags (trace, span) {
     registerExtraService(tags['service.name'])
   }
 
-  for (const tag in tags) {
+  for (const [tag, value] of Object.entries(tags)) {
+    // TODO(BridgeAR)[31.03.2025]: Check how many tags are defined in average.
```

Comment:
Hard to come up with an actual number for this, but I'd say 10-20 would probably be pretty standard (meta and metrics combined)

Reply:
Sounds good. In that case doing the lookup first is faster (right now each tag has to go through all cases, so we have half of all special cases in average in case of a equal distribution. I guess the special ones are actually less frequent than the last entry, so it's even worse. If we do the special handling up front once, we only need to pay the lookup cost once for each case. That would already be the case with a single or definitely two tags. Then afterwards it's a constant single lookup per tag instead of all special cases).